src/xeus_calc_interpreter.cpp

```diff
@@ -50,6 +50,8 @@ namespace xeus_calc
     {
         const std::string ops = "-+/*^";
         std::stringstream ss;
+        std::string characters = ".()";
+        std::vector<char> parenthesis;
```

Comment:
Using an `std::vector<char>` feels bad here, you should maybe keep a parenthesis counter instead? Maybe an `std::size_t` that you increment when you found an opening parenthesis and decrement when you find a closing one?